log-output/pkg/index/service.go

```diff
@@ -21,27 +23,53 @@ func NewIndexService() *Service {
 }
 
 func (s Service) ListLogs(w http.ResponseWriter, r *http.Request) {
-	var err error
-	var logs [][]byte
 	var indexName = mux.Vars(r)["index_name"]
 
-	s.m.RLock()
+	if r.Header.Get("Accept") == "application/json" {
+		var logs []map[string]interface{}
+
+		s.m.RLock()
+		// check if this is a valid index
+		if _, exists := s.indices[indexName]; exists {
+			for _, log := range s.indices[indexName].Logs {
```

Comment:
Can you copy logs out into a variable so we don't block for the json unmarshall?